flang/lib/Semantics/scope.cpp

```diff
@@ -374,6 +374,25 @@ bool Scope::IsParameterizedDerivedType() const {
   return false;
 }
 
+bool Scope::IsKindParameterizedDerivedType() const {
```

Comment:
The name of this function doesn't read quite right to me.  How about "IsDerivedTypeWithKindParameter"?